substrate/utils/wasm-builder/src/wasm_project.rs

```diff
@@ -775,12 +814,15 @@ fn offline_build() -> bool {
 }
 
 /// Build the project and create the bloaty runtime blob.
+///
+/// Returns the path to the generated bloaty runtime blob.
 fn build_bloaty_blob(
 	target: RuntimeTarget,
 	blob_build_profile: &Profile,
 	project: &Path,
 	default_rustflags: &str,
 	cargo_cmd: CargoCommandVersioned,
+	#[cfg(feature = "metadata-hash")] metadata_hash: Option<[u8; 32]>,
```

Comment:
Why do you need to feature gate the existence of the arg instead of just setting it to `None` when the feature is disabled?

Reply:
First I had it this way, but then I was doing feature gating all the other things and have done it here as well.